pgxn/neon/walproposer_pg.c

```diff
@@ -135,13 +137,21 @@ PGDLLEXPORT void
 WalProposerSync(int argc, char *argv[])
 {
 	WalProposer *wp;
+	MemoryContext ctx;
+	/* contains copies of GUCs and other runtime allocations */
+	WalPropMemCtx = AllocSetContextCreate(TopMemoryContext,
```

Comment:
Most likely walproposer*.c code never leaves top memory context, so I doubt having separate one is useful.